src/context/fetcher/TenantBillingDetails.tsx

```diff
@@ -66,17 +66,28 @@ export const useTenantBillingDetails = () => {
     return useContext(TenantContext);
 };
 
-export const useTenantUsesExternalPayment = (selectedTenant: string) => {
+export const useTenantUsesExternalPayment = (
+    selectedTenant: string
+): [boolean, string | null] => {
     const { tenantBillingDetails } = useTenantBillingDetails();
 
     return useMemo(() => {
-        return !tenantBillingDetails
-            ? false
-            : tenantBillingDetails.some((tenantBillingDetail) => {
-                  return (
-                      tenantBillingDetail.tenant === selectedTenant &&
-                      tenantBillingDetail.payment_provider === 'external'
-                  );
-              });
+        if (!tenantBillingDetails) {
+            return [false, null];
+        }
+
+        const selectedTenantDetails = tenantBillingDetails.find(
+            (tenantBillingDetail) =>
+                tenantBillingDetail.tenant === selectedTenant
+        );
+
+        const marketplaceProvider = selectedTenantDetails?.gcm_account_id
+            ? 'gcp'
+            : null; //add aws support
```

Comment:
Is the comment on line 86 a future TODO or in scope for this branch? If the former, could you mark it as such?